MusicApp.SongService.Web/Controllers/SongsController.cs

```diff
@@ -50,7 +50,9 @@ public async Task<IActionResult> CreateSong(SongInputDto songInputDto, Cancellat
         var createCommand = new CreateSongCommand(songInputDto, artist);
         var song = await _mediator.Send(createCommand, cancellationToken);
 
-        return Created($"/api/songs/{song.Id}", song);
+        var actionName = nameof(GetSongById);
+
+        return CreatedAtAction(actionName, song);
```

Comment:
Why do you use createdAtAction?

Reply:
Because it writes uri to the created object itself.